### DISCUSSION TITLE
Version flag should only output if no commands are called.

### DISCUSSION THREAD

**Discussion by surgiie** [bashly-framework/bashly#197] [Ideas]
I have a command that proxies to a 3rd party/external cli, that cli has it's own `--version` option. If i would like to print out the `--version`
of that tool via my proxy command:

`mycli external-cli --version`  

It output's my cli version instead of the external cli im proxying a command to.

It would make more sense to only print out `--version` if the cli is called with no argments. i.e `mycli --version`

**Comment by DannyBen** [comment]
Can you provide a minimal bashly.yml to reproduce?

**Comment by DannyBen** [comment]
I think I fixed the reason for this happening. Subcommands should not handle `--version` at all.

I am not sure if you are using the `delegate` directive, or not, but a fix is in #198

**Comment by surgiie** [comment]
Away from my pc the next few days, so unable to provide a bashly.yml file, but i am not using `delegate` directive.

**Reply by DannyBen** [reply]
Alright. I will wait a little more before merging, in case you want to review the PR or even test it with your use case.
I believe it should allow you to do what you want, but I can't be sure without your confirmation or reproduction case.

**Reply by surgiie** [reply]
Appears to be working, feel free to release whenever.

**Reply by DannyBen** [reply]
Thanks. Merged and will release soon.

**Comment by DannyBen** [comment]
Version 0.7.8 was released with this fix.
